### PR TITLE
doc: add note to described the possibility to use MultiNamespacedCach…

### LINKED ISSUE #2008: Add MultiNamespacedCacheBuilder option in the docs which was introduced with the #1512 and is valid just in the master branch.
Add MultiNamespacedCacheBuilder option in the docs which was introduced with the https://github.com/operator-framework/operator-sdk/pull/1512 and is valid just in the master branch. 

```go
var namespaces []string // List of Namepsaces
// Create a new Cmd to provide shared dependencies and start components
mgr, err := manager.New(cfg, manager.Options{
   NewCache: cache.MultiNamespacedCacheBuilder(namespaces),
   MapperProvider:     restmapper.NewDynamicRESTMapper,
   MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
})
```

- Update: https://github.com/operator-framework/getting-started#manager
- https://github.com/operator-framework/operator-sdk/blob/05a106abd9ff94bb2f0cc84b5054d0a603c9474f/doc/user-guide.md#L64

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Add a note to redirect the user to check in the controller-runtime docs the possibility to use MultiNamespacedCache for watch/cache N namespaces. 

**Motivation for the change:**
Note: If this PR is fixing an issue make sure to add a note saying:
Closes #2008

### CODE REVIEW

File: doc/user-guide.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ By default this will be the namespace that the operator is running in. To watch
 mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
 ```
 
+**NOTE:** See [here](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) that for some scenarios is possible to inform a list of namespaces as well.
```
Comment (joelanford) [inline doc/user-guide.md]:
```suggestion
It is also possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
```

And then perhaps include an example of how someone could watch 2 namespaces?

**Comment by camilamacedo86** [comment]
Hi @joelanford,

removed the hold since it the  kubernetes-sigs/controller-runtime#701 is merged.

File: doc/user-guide.md [hunk 2]
```diff
@@ -63,6 +63,17 @@ By default this will be the namespace that the operator is running in. To watch
 mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
 ```
 
+It is also possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
+```Go
+var namespaces []string // List of Namepsaces
```
Comment (camilamacedo86) [inline doc/user-guide.md]:
```suggestion
var namespaces []string // List of Namespaces
```

File: doc/user-guide.md [hunk 3]
```diff
@@ -63,6 +63,17 @@ By default this will be the namespace that the operator is running in. To watch
 mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
 ```
 
+It is also possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
```
Comment (hasbro17) [inline doc/user-guide.md]: We're using [reference style links](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) throughout the rest of the user-guide to make the raw markdown easier to read.
So perhaps change this to a reference link.

File: doc/user-guide.md (hunk 2 above)
Comment (hasbro17) [inline doc/user-guide.md]: Still a typo here:
```suggestion
var namespaces []string // List of Namespaces
```

File: doc/user-guide.md (hunk 3 above)
Comment (camilamacedo86) [reply doc/user-guide.md]: done

File: doc/user-guide.md (hunk 2 above)
Comment (camilamacedo86) [reply doc/user-guide.md]: Done

File: doc/user-guide.md [hunk 4]
```diff
@@ -63,6 +63,17 @@ By default this will be the namespace that the operator is running in. To watch
 mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
 ```
 
+It is also possible to use the [MultiNamespacedCacheBuilder][multi-namespaced-cache-builder] to watch a specific set of namespaces:
+```Go
+var namespaces []string // List of Namespaces
+// Create a new Cmd to provide shared dependencies and start components
+mgr, err := manager.New(cfg, manager.Options{
+   NewCache: cache.MultiNamespacedCacheBuilder(namespaces),
+   MapperProvider:     restmapper.NewDynamicRESTMapper,
```
Comment (joelanford) [inline doc/user-guide.md]: Now that https://github.com/kubernetes-sigs/controller-runtime/pull/554 is merged and we've updated to v0.4.0, we should remove our dynamic rest mapper implementation and any references to it.

That can be done in a follow up though.

**Pull request merged**